src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2904,12 +2934,25 @@ def aks_agentpool_scale(cmd, client, resource_group_name, cluster_name,
 
 
 def aks_agentpool_upgrade(cmd, client, resource_group_name, cluster_name,
-                          kubernetes_version,
                           nodepool_name,
+                          kubernetes_version='',
+                          node_image_only=False,
                           no_wait=False):
     instance = client.get(resource_group_name, cluster_name, nodepool_name)
```

Comment:
These 2 lines can be moved below the `node_image_only` part.

Reply:
make sense. moved!